### ISSUE TITLE
[HL:S] Glock 17 alt-fire doesn't work while underwater

### ISSUE THREAD

**Issue by Yetoo1** [ValveSoftware/Source-1-Games#2987]
Glock 17 alt-fire doesn't work while underwater. In goldsrc it does work.

HL:S Version:
Build Label: 5377866 # Uniquely identifies each build
Network PatchVersion: 5377866 # Determines client and server compatibility
Protocol version: 24 # High level network protocol version

System info here:
https://web.archive.org/web/20191207232514/https://pastebin.com/qJ5SPXg3

**Comment by 1157** [comment]
https://half-life.fandom.com/wiki/Glock_17

The rapid-fire mode cannot be utilized when underwater.

**Comment by Yetoo1** [comment]
> https://half-life.fandom.com/wiki/Glock_17
> 
> The rapid-fire mode cannot be utilized when underwater.

Well, I tested goldsrc and Glock 17 alt-fire was able to fire underwater. It's either a bug with Half-Life or the fan page is incorrect.

**Comment by Koterminus** [comment]
@1157 The Fandom version of the wiki is outdated and unmoderated, I would take anything said there with a massive grain of salt. Use the [Combine OverWiki](https://combineoverwiki.net/wiki/Main_Page) instead.

I just tested this issue and @Yetoo1 is right, in Half-Life 1 the Glock 17 can be fired underwater with the alt-fire button in both singleplayer and multiplayer, but in Half-Life: Source it cannot.

**Comment by Yetoo1** [comment]
I have a Sierra disk version of Half-Life that I can verify on, but it's going to take a bit for me to find it.

**Comment by 1157** [comment]
The page is correct.    We don't have gibs  or the rpg laser dot thats 250 ms behind the target. Some things should not be "like Goldsrc" Who's to say should it fire or not underwater 2 different engines.

**Comment by Yetoo1** [comment]
> The page is correct.

Why is the page correct?

**Comment by 1157** [comment]
The page i linked

https://half-life.fandom.com/wiki/Half-Life:_Source

linked to the Glock page that refers to Half life Source that Alt fire does not fire underwater.

https://half-life.fandom.com/wiki/Glock_17

I did find under shotgun.cpp  for half life that it's not allowed to fire underwater
`BOOL CShotgun::Deploy( )
{
	return DefaultDeploy( "models/v_shotgun.mdl", "models/p_shotgun.mdl", SHOTGUN_DRAW, "shotgun" );
}

void CShotgun::PrimaryAttack()
{
	// don't fire underwater
	if (m_pPlayer->pev->waterlevel == 3)
	{
		PlayEmptySound( );
		m_flNextPrimaryAttack = GetNextAttackDelay(0.15);
		return;
	}

	if (m_iClip <= 0)
	{
		Reload( );
		if (m_iClip == 0)
			PlayEmptySound( );
		return;
	}
This same bit of code is missing for the glock so it _can`_ fire underwater.

**Comment by Yetoo1** [comment]
1. Where does it specifically say in the links that you provided that Half-Life: Source can't fire alt-fire underwater? Before you say that it was linked from the Half-Life: Source page and therefore must be an article for the Half-Life: Source pistol (which it isn't), can you find me a Wikia entry for the Glock 17 just for the Half-Life game within that fandom site?
2. With that code and context, you have affirmed my findings that that alt-fire does work in the original Half-Life. 
3. I tested pistol alt-fire with the following version of Half-Life:
Build 742
Exe: 15:02:21 Nov 5 1998 

I can confirm alt-fire does work underwater in this first official release build of Half-Life.

**Comment by Yetoo1** [comment]
So the fandom page must be wrong and must clarify which version of the game it doesn't work with due to it being a bug.

**Comment by 1157** [comment]
Yes you are correct your 21 year old non-Source half-life copy  glock does alt-fire underwater, 
Ever thought that maybe is was removed with the Source version.Happy issue reporting.

**Comment by Koterminus** [comment]
There's no reason for this feature to be missing from the Source version, so it should be added in (or fixed, if it existed in an earlier build and was broken or removed by a later update).
